### PR TITLE
Fix VM manifest rendering in export-controller

### PR DESCRIPTION
**Pull request by alromeros**
### What this PR does
**Before this PR:**

Since https://github.com/kubevirt/kubevirt/pull/8953, the export controller allows rendering virtual machine manifests to complete VM information with export URLs. However, this mechanism uses custom logic to complete the `DataVolumeTemplate` field, leading to api incompatibilities when the original VM had volumes with populated SourceRef fields.

**After this PR:**

This pull request aims to fix this behavior by setting the `SourceRef` field to nil when populating the HTTP source in the export controller.

Fixes # https://issues.redhat.com/browse/CNV-40606

### Why we need it and why it was done in this way

I considered modifying the manifest in the export sever (instead of in the controller). However, since the controller is setting the http source, I think it makes way more sense to fix this in the controller.

### Release note

```release-note
Bugfix: Fix VM manifest rendering in export controller
```

### CODE REVIEW

File: pkg/storage/export/export/export_test.go [hunk 1]
```diff
@@ -1288,6 +1289,11 @@ var _ = Describe("Export controller", func() {
 					Source: &cdiv1.DataVolumeSource{
 						Blank: &cdiv1.DataVolumeBlankImage{},
 					},
+					SourceRef: &cdiv1.DataVolumeSourceRef{
+						Kind:      "",
+						Name:      "test",
+						Namespace: ptr.To("default"),
```
Comment (alicefr) [inline pkg/storage/export/export/export_test.go:1295]: Can you please use the kubevirt pointer?

Comment (alromeros) [reply pkg/storage/export/export/export_test.go:1295]: Sure!

**Review by ShellyKa13 — COMMENTED** [review]
/lgtm

**Pull request merged**